### PR TITLE
update: improve getting platform value from operator

### PR DESCRIPTION
**Pull request by zdtsw**
mainly cherrypick change from https://issues.redhat.com/browse/RHOAIENG-8898 + updates to keep it for 2.8 code base

Many thanks for submitting your Pull Request ❤️!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md)
- [ ] Pull Request contains description of the issue
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request

## JIRA issue:

https://issues.redhat.com/browse/RHOAIENG-8898
## Merge criteria:

- [ ] Have a meaningful commit messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pkg/deploy/setup.go [hunk 1]
```diff
@@ -23,62 +20,40 @@ const (
...
 
-// isManagedRHODS checks if CRD add-on exists and contains string ManagedRhods.
-func isManagedRHODS(cli client.Client) (Platform, error) {
-	catalogSourceCRD := &apiextv1.CustomResourceDefinition{}
+// isManagedRHODS checks if catsrc CR add-on exists ManagedRhods.
+func isManagedRHODS(ctx context.Context, cli client.Client) (Platform, error) {
+	catalogSource := &ofapi.CatalogSource{}
 
-	err := cli.Get(context.TODO(), client.ObjectKey{Name: "catalogsources.operators.coreos.com"}, catalogSourceCRD)
+	err := cli.Get(ctx, client.ObjectKey{Name: "addon-managed-odh-catalog", Namespace: "openshift-marketplace"}, catalogSource)
```
Comment (bdattoma) [inline pkg/deploy/setup.go:47]: cannot we use some global variables to set the namespace and catalog source name everywhere?

File: components/datasciencepipelines/datasciencepipelines.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ func (d *DataSciencePipelines) ReconcileComponent(ctx context.Context,
 	enabled := d.GetManagementState() == operatorv1.Managed
 	monitoringEnabled := dscispec.Monitoring.ManagementState == operatorv1.Managed
 
-	platform, err := deploy.GetPlatform(cli)
+	platform, err := deploy.GetPlatform(ctx, cli)
```
Comment (VaishnaviHire) [inline components/datasciencepipelines/datasciencepipelines.go:75]: Do we need this change? I think we should minimize the changes going into backport

Comment (zdtsw) [reply components/datasciencepipelines/datasciencepipelines.go:75]: so the change is in the `isManagedRHODS()` which now require a ctx to do `cli.Get()`
then either in the caller (`GetPlatform()`)  pass down ctx or we have a `context.TODO` in `isManagedRHODS()`

I am open for either way.

Comment (zdtsw) [reply components/datasciencepipelines/datasciencepipelines.go]: I did a revert on the context part, now it should be much small updates in the PR @VaishnaviHire

Comment (VaishnaviHire) [reply components/datasciencepipelines/datasciencepipelines.go]: Hi thanks ! Yes this looks good to me now

**Review by VaishnaviHire — APPROVED** [review]
/lgtm

**Pull request merged**